crates/sui-indexer-alt-framework/src/pipeline/concurrent/commit_watermark.rs

```diff
@@ -128,7 +128,7 @@ pub(super) fn commit_watermark<H: Handler + 'static>(
 
                             // This is the next checkpoint -- include it.
                             Ordering::Equal => {
-                                watermark = pending.remove().watermark;
+                                watermark = Some(pending.remove().watermark);
                                 watermark_needs_update = true;
```

Comment:
Can we use the `watermark.is_some()` to mean `watermark_needs_update`, and if so, can we rejig the rest of the code below to move the metrics updates underneath that condition? That might help simplify the control flow, meaning you don't need to worry about moving the other channel admin work around.